### PR TITLE
Make MD5 algorithm replaceable with SHA1

### PR DESCRIPTION
**Pull request by Scooletz**
Solves #25

### CODE REVIEW

**Comment by Scooletz** [comment]
Ping @Particular/azure-maintainers Ready for a review & merge

File: src/Transport/Shortener.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        }
+
+        public static string Sha1(string test)
+        {
+            using (var provider = new SHA1CryptoServiceProvider())
+            {
+                var inputBytes = Encoding.Default.GetBytes(test);
+                var hashBytes = provider.ComputeHash(inputBytes);
+
+                return ToChars(hashBytes);
```
Comment (SeanFeldman) [inline src/Transport/Shortener.cs]: If you'd like to shorten, this would work: `return string.Join("", hashBytes.Select(x => x.ToString("x2")));`

Comment (Scooletz) [inline src/Transport/Shortener.cs]: This would make so many allocs. Can we keep it the way it is?

Comment (yvesgoeleven) [inline src/Transport/Shortener.cs]: Sure

**Pull request merged**